app/packages/core/src/plugins/SchemaIO/components/DynamicIO.tsx

```diff
@@ -66,6 +55,44 @@ function useCustomComponents() {
   }, {});
 }
 
+// todo: need to improve initializing the state... refactor this function
+function useStateInitializer(props: ViewPropsType) {
+  const { data, schema, onChange, path, root_id } = props;
+  const computedSchema = getComputedSchema(props);
+  const { default: defaultValue } = computedSchema;
+  const basicData = useMemo(() => {
+    if (!isCompositeView(schema)) {
+      return data;
```

Comment:
Does this intentional return undefined otherwise...?

Reply:
Yah, below I have a check to ignore undefined `basicData`. The intention here is to only initialize non-composite data similar to initialization of default.